pkg/rbeconfigsgen/rbeconfigsgen.go

```diff
@@ -0,0 +1,847 @@
+	}
+	log.Fatalf("Invalid OS: %q", os)
+	return "<invalid os>", "<invalid os>"
+}
+
+func newDockerRunner(containerImage string, stopContainer bool) (*dockerRunner, error) {
+	if containerImage == "" {
+		return nil, fmt.Errorf("container image was not specified")
+	}
+	d := &dockerRunner{
+		containerImage: containerImage,
+		stopContainer:  stopContainer,
+		dockerPath:     "docker",
+	}
+	if _, err := runCmd(d.dockerPath, "pull", d.containerImage); err != nil {
+		return nil, fmt.Errorf("docker was unable to pull the toolchain container image %q: %w", d.containerImage, err)
+	}
+	resolvedImage, err := runCmd(d.dockerPath, "inspect", "--format={{index .RepoDigests 0}}", d.containerImage)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert toolchain container image %q into a fully qualified image name by digest: %w", d.containerImage, err)
+	}
+	resolvedImage = strings.TrimSpace(resolvedImage)
+	log.Printf("Resolved toolchain image %q to fully qualified reference %q.", d.containerImage, resolvedImage)
+	d.resolvedImage = resolvedImage
+
+	cid, err := runCmd(d.dockerPath, "create", "--rm", d.resolvedImage, "sleep", "infinity")
+	if err != nil {
+		return nil, fmt.Errorf("failed to create a container with the toolchain container image: %w", err)
+	}
+	cid = strings.TrimSpace(cid)
+	if len(cid) != 64 {
+		return nil, fmt.Errorf("container ID %q extracted from the stdout of the container create command had unexpected length, got %d, want 64", cid, len(cid))
+	}
+	d.containerID = cid
+	log.Printf("Created container ID %v for toolchain container image %v.", d.containerID, d.resolvedImage)
+	if _, err := runCmd(d.dockerPath, "start", d.containerID); err != nil {
+		return nil, fmt.Errorf("failed to run the toolchain container: %w", err)
+	}
+	return d, nil
+}
+
+// execCmd runs the given command inside the docker container and returns the output with whitespace
+// trimmed from the edges.
+func (d *dockerRunner) execCmd(args ...string) (string, error) {
+	a := []string{"exec"}
+	if d.workdir != "" {
+		a = append(a, "-w", d.workdir)
+	}
+	for _, e := range d.env {
+		a = append(a, "-e", e)
+	}
+	a = append(a, d.containerID)
+	a = append(a, args...)
+	o, err := runCmd(d.dockerPath, a...)
+	return strings.TrimSpace(o), err
+}
+
+func (d *dockerRunner) cleanup() {
+	if !d.stopContainer {
+		log.Printf("Not stopping container %v of image %v because the Cleanup option was set to false.", d.containerID, d.resolvedImage)
+		return
+	}
+	if _, err := runCmd(d.dockerPath, "stop", "-t", "0", d.containerID); err != nil {
+		log.Printf("Failed to stop container %v of toolchain image %v but it's ok to ignore this error if config generation & extraction succeeded.", d.containerID, d.resolvedImage)
+	}
+}
+
+// copyTo copies the local file at 'src' to the container where 'dst' is the path inside the
+// container. d.workdir has no impact on this function.
+func (d *dockerRunner) copyTo(src, dst string) error {
```

Comment:
I think calling this function `CopyToContainer` makes it more obvious that the `src` is outside of the container and the `dst` is in the container.

Similarly replace `copyFrom` with `copyFromContainer`.